tests/integration/steps/integration_steps.go

```diff
@@ -0,0 +1,699 @@
+	}
+}
+
+// Medusa related functions
+func ICreateTheMedusaSecretInTheNamespaceApplyingTheFileStep(secretFile string) {
+	By(fmt.Sprintf("I create the Medusa secret applying the %s secret file", secretFile))
+
+	home, _ := os.UserHomeDir()
+	medusaSecretPath, _ := filepath.Abs(strings.Replace(secretFile, "~", home, 1))
+	kubectlOptions := k8s.NewKubectlOptions("", "", namespace)
+	k8s.KubectlApply(GinkgoT(), kubectlOptions, medusaSecretPath)
+}
+
+func IPerformABackupWithMedusaNamedStep(backupName string) {
+	By(fmt.Sprintf("I perform a backup with Medusa named %s", backupName))
+
+	kubectlOptions := k8s.NewKubectlOptions("", "", namespace)
+	backupChartPath, err := filepath.Abs("../../charts/backup")
+	if err != nil {
+		Fail(fmt.Sprintf("Couldn't find the absolute path for backup charts"))
+	}
+
+	helmOptions := &helm.Options{
+		SetValues: map[string]string{
+			"name":                     backupName,
+			"cassandraDatacenter.name": "dc1",
+		},
+		KubectlOptions: k8s.NewKubectlOptions("", "", namespace),
+	}
+	helm.Install(GinkgoT(), helmOptions, backupChartPath, "test")
+
+	// Wait for the backup to be finished
+	// kubectl get cassandrabackup backup1 -o jsonpath='{.status.finished}' -n k8ssandra2021021306435807
+	attempts := 0
+	for {
+		attempts++
+		output, err := k8s.RunKubectlAndGetOutputE(GinkgoT(), kubectlOptions, "get", "cassandrabackup", backupName, "-o", "jsonpath='{.status.finished}'")
+		if err == nil && len(output) > 0 {
+			var nodes []string
+			json.Unmarshal([]byte(strings.ReplaceAll(output, "'", "")), &nodes)
+			if len(nodes) == 1 {
+				return
+			}
+		}
+		if attempts > 12 {
+			Fail(fmt.Sprintf("Backup didn't succeed within timeout: %s", err))
+		}
+		time.Sleep(10 * time.Second)
+	}
+}
+
+func IRestoreTheBackupNamedUsingMedusaStep(backupName string) {
+	By(fmt.Sprintf("I restore the backup with Medusa named %s", backupName))
+
+	restoreChartPath, err := filepath.Abs("../../charts/restore")
+	if err != nil {
+		Fail(fmt.Sprintf("Couldn't find the absolute path for restore charts"))
+	}
+
+	helmOptions := &helm.Options{
+		SetValues: map[string]string{
+			"backup.name":              backupName,
+			"cassandraDatacenter.name": "dc1",
+			"name":                     "restore-test",
+		},
+		KubectlOptions: k8s.NewKubectlOptions("", "", namespace),
+	}
+	helm.Install(GinkgoT(), helmOptions, restoreChartPath, "restore-test")
+	// Give a little time for the cassandraDatacenter resource to be recreated
+	time.Sleep(60 * time.Second)
+	WaitForPodWithLabelToBeReady("app.kubernetes.io/managed-by=cass-operator", 30*time.Second, 10)
+}
+
+// Reaper related steps
+func ICanCheckThatAClusterNamedWasRegisteredInReaperInNamespaceStep(clusterName string) {
+	By(fmt.Sprintf("I can check that a cluster named %s was registered in Reaper", clusterName))
+
+	restClient := resty.New()
+	attempts := 0
+	for {
+		attempts++
+		response, err := restClient.R().Get("http://repair.localhost:8080/cluster")
```

Comment:
We have reaperClient in reaper-operator which you can use.

Reply:
And if it lacks some feature you need for testing, add there.

Reply:
Neat, I didn't know about this client. Will definitely use it instead.